crates/astria-sequencer-types/src/sequencer_block_data.rs

```diff
@@ -167,35 +217,24 @@ impl SequencerBlockData {
             return Err(Error::HashOfHeaderBlockHashMismatach);
         }
 
-        let data_hash: [u8; 32] = header
-            .data_hash
-            .ok_or(Error::MissingDataHash)?
-            .as_bytes()
-            .try_into()
-            .map_err(Error::DataHashWrongLength)?;
+        let Some(Hash::Sha256(data_hash)) = header.data_hash else {
+            // header.data_hash is Option<Hash> and Hash itself has
+            // variants Sha256([u8; 32]) or None.
+            return Err(Error::MissingDataHash);
+        };
 
         let action_tree_root_hash = sha2::Sha256::digest(action_tree_root);
         if !action_tree_root_inclusion_proof.verify(&action_tree_root_hash, data_hash) {
             return Err(Error::ActionTreeRootVerification);
         }
 
-        let chain_ids_commitment_hash = sha2::Sha256::digest(chain_ids_commitment);
```

Comment:
note: all of this is rolled into `assert_chain_ids_are_included` because that's used elsewhere.